translators/ChatGPTTranslator.js

```diff
@@ -0,0 +1,78 @@
+class ChatGPTTranslator {
+    constructor(model = 'gpt-4o', openApiKey = "") {
+        this.apiKey = openApiKey;
+        this.apiUrl = 'https://api.openai.com/v1/chat/completions';
+        this.model = model;
+    }
+
+    async translate(text, from, to) {
+        if (!text) return '';
+
+        const prompt = (from === 'auto' || from === '')
+            ? `Translate text to ${to}. Only return the translation without any additional text:\n\n${text}`
+            : `Translate text from ${from} to ${to}. Only return the translation without any additional text:\n\n${text}`;
+
+        try {
+            const response = await fetch(this.apiUrl, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': `Bearer ${this.apiKey}`
+                },
+                body: JSON.stringify({
+                    model: this.model,
+                    messages: [{ role: 'user', content: prompt }],
+                    temperature: 0.3
+                })
+            });
+
+            const data = await response.json();
+            
+            if (!data.choices || !data.choices[0]?.message?.content) {
+                console.error("API error:", data);
+                return '';
+            }
+
+            return data.choices[0].message.content.trim();
+
+        } catch (error) {
+            console.error("Translation request failed:", error);
+            return '';
```

Comment:
If exception has occurs, we must throw an error, to handle it properly. Otherwise user will see empty text in translations and will be confused